src/wallet/rpcdump.cpp

```diff
@@ -572,7 +576,9 @@ UniValue dumpwallet(const JSONRPCRequest& request)
     EnsureWalletIsUnlocked();
 
     ofstream file;
-    file.open(request.params[0].get_str().c_str());
+    boost::filesystem::path filepath = request.params[0].get_str();
+    filepath = absolute(filepath);
```

Comment:
> Maybe an absolute path or if not, relative to the data-dir? If possible?

Good idea. It's our (effective) convention for command line arguments as well. I think it would be good to `chdir` the daemon to the data directory at start. This also avoids issues with the daemon holding on to resources (e.g. the starting directory could be deleted later) discussed in #8278.

>  RPC may even be called on a different machine

A way to dump the wallet over the network could be useful. The practical problem is that there is currently no way to do streaming writes to the network from RPC code, so the entire wallet would have to be dumped in memory, which would result in huge memory and parsing/generating overhead. I looked at streaming to the HTTP in a different context here: #7759.

Also yes there's a bit of a security concern though it's no different from dumpprivkey and friends...